parser/match_operations.go

```diff
@@ -67,11 +67,11 @@ func (s AmountSign) Match(amount *types.Amount) bool {
 		return true
 	}
 
-	if s == PositiveOrZeroAmountSign && (numeric.Sign() == 1 || amount.Value == "0") {
+	if s == PositiveOrZeroAmountSign && (numeric.Sign() == 1 || numeric.Cmp(big.NewInt(0)) == 0) {
```

Comment:
We can take a look at https://stackoverflow.com/questions/64257065/is-there-another-way-of-testing-if-a-big-int-is-0
bits compare is proven to be 20 times faster than comparing it to another big.Int representing 0

Reply:
Great comment @xiaying-peng ! Checked and updated PR.